R/EventStudy.R

```diff
@@ -170,56 +174,89 @@ EventStudy <- function(estimator, data, outcomevar, policyvar, idvar, timevar, c
     n_unique_rows <- nrow(data[!base::duplicated(data_ids),])
     if (n_unique_rows != n_units*n_periods) {
         warning("Dataset is unbalanced.")
+        unbalanced <- TRUE
+    } else {
+        unbalanced <- FALSE
+    }
+
+    data.table::setorderv(data, c(idvar, timevar))
+
+    detect_holes <- function(dt, idvar, timevar) {
+        dt <- data.table::as.data.table(dt)
+        holes_per_id <- dt[, .SD[!is.na(base::get(timevar))], by = c(idvar)
+                         ][, list(holes = any(base::diff(base::get(timevar)) != 1)), 
+                            by = c(idvar)]
+        
+        return(any(holes_per_id$holes))
```

Comment:
If gaps are detected should we issue a warning like

"Note: gaps of more than one in the time variable are detected. Treating these as gaps in the panel."